Libraries/MiscDrivers/PushButton/pb.c

```diff
@@ -74,18 +75,26 @@ int PB_RegisterCallback(unsigned int pb, pb_callback callback)
     return E_NO_ERROR;
 }
 
-//******************************************************************************
-void GPIO0_IRQHandler(void)
-{
-    MXC_GPIO_Handler(MXC_GPIO_GET_IDX(MXC_GPIO0));
-}
-void GPIO1_IRQHandler(void)
-{
-    MXC_GPIO_Handler(MXC_GPIO_GET_IDX(MXC_GPIO1));
-}
-void GPIO2_IRQHandler(void)
+/******************************************************************************/
+int PB_RegisterCallbackRiseFall(unsigned int pb, pb_callback callback)
 {
-    MXC_GPIO_Handler(MXC_GPIO_GET_IDX(MXC_GPIO2));
+    MXC_ASSERT(pb < num_pbs);
```

Comment:
Instead of asserting, you can also return an error code if incorrect `pb` parameter, so this function can follow the doxygen comments and not have one return code.